ext/pdo/pdo_stmt.c

```diff
@@ -1747,7 +1747,7 @@ int pdo_stmt_setup_fetch_mode(pdo_stmt_t *stmt, zend_long mode, zval *args, uint
 	flags = mode & PDO_FETCH_FLAGS;
 	retval = pdo_stmt_verify_mode(stmt, mode, 0);
 
-	if (FAILURE == retval) {
+	if (false == retval) {
```

Comment:
Please just check `if (!pdo_stmt_verify_mode())` here instead. Right now you're storing boolean into int `retval`.

Reply:
Opened #6215 for it as it also affects PHP 7.x